tests/simulation_framework/test_cache_model.py

```diff
@@ -0,0 +1,71 @@
+import json
+import os
+import tempfile
+from typing import Optional
+
+import pydantic as pd
+
+import flow360.component.simulation.units as u
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+from flow360.component.simulation.framework.cache_model_base import CacheModelBase
+from flow360.component.simulation.unit_system import (
+    DensityType,
+    LengthType,
+    TemperatureType,
+)
+
+
+class TempThermalStateCache(Flow360BaseModel):
+    altitude: Optional[LengthType.Positive] = None
+    temperature_offset: Optional[TemperatureType] = None
+
+
+class TempThermalState(CacheModelBase):
+    temperature: TemperatureType.Positive = pd.Field(288.15 * u.K, frozen=True)
+    density: DensityType.Positive = pd.Field(1.225 * u.kg / u.m**3, frozen=True)
+    _cached: TempThermalStateCache = TempThermalStateCache()
+
+    @classmethod
+    def from_standard_atmosphere(
+        cls, altitude: LengthType.Positive = 0 * u.m, temperature_offset: TemperatureType = 0 * u.K
+    ):
+        density = 1.225 * u.kg / u.m**3
+        temperature = 288.15 * u.K
+
+        state = cls(
+            density=density,
+            temperature=temperature,
+        )
+        state._cached = TempThermalStateCache(
+            altitude=altitude, temperature_offset=temperature_offset
+        )
+        return state
+
+    @property
+    def altitude(self) -> Optional[LengthType.Positive]:
+        return self._cached.altitude
+
+
+class TempOperatingCondition(Flow360BaseModel):
+    thermal_state: TempThermalState
+    some_value: float
+
+
+def test_cache_model():
+    operating_condition = TempOperatingCondition(
+        some_value=1230,
+        thermal_state=TempThermalState.from_standard_atmosphere(altitude=100 * u.m),
+    )
+    with tempfile.NamedTemporaryFile(mode="w", suffix=".json", delete=False) as temp_file:
+        temp_file_name = temp_file.name
+        operating_condition.to_file(temp_file_name)
+
+    try:
+        with open(temp_file_name) as fp:
+            model_dict = json.load(fp)
+            assert model_dict["some_value"] == 1230
+            assert model_dict["thermal_state"]["_cached"]
```

Comment:
can you instead of assert here, check the dict is exactly the same as a handwritten one?

Reply:
or you can compare the two objects. The original one and the one loaded from JSON. Similar to what we do with SimulationParam seralizer.

Reply:
@pytest.mark.usefixtures("array_equality_override")